test/unit/src/index-paginator-test.mjs

```diff
@@ -0,0 +1,85 @@
+import { join } from 'node:path'
+import process from 'node:process'
+// import qs from 'node:querystring'
+import test from 'tape'
+import { copy, defaults, resetServer as reset, server } from '../../lib/index.mjs'
+// import url from 'node:url'
+
+let client
+let { config, service } = defaults
+let jsonHeaders = { 'content-type': 'application/json' }
+// let xmlHeaders = { 'content-type': 'application/xml' }
+
+
+const simpleResponseBodies = [
+  {
+    Token: 't1',
+    Accumulator: 'a1',
+  },
+  {
+    Token: 't2',
+    Accumulator: 'a2',
+  },
+  {
+    Accumulator: 'a3',
+  },
+]
+
+
+test('Set up env', async t => {
+  t.plan(2)
+  let cwd = process.cwd()
+  let sut = 'file://' + join(cwd, 'src', 'index.js')
+  client = (await import(sut)).default
+  t.ok(client, 'aws-lite client is present')
+  let started = await server.start()
+  t.ok(started, 'Started server')
+})
+
+test('Async Iterator - raw client', async t => {
+  t.plan(6)
+  let aws = await client(config)
+  let headers = copy(jsonHeaders)
+  let expectedUrl
+  let page
+  let response
+  let request
+
+  // Returns async iterator
+  response = await aws({ service, headers, paginate: 'iterator', paginator: { cursor: 'Cursor', token: 'Token', type: 'query' }, query: { } })
+
+  // Query type, first page
+  server.use({ responseBody: simpleResponseBodies[0], responseHeaders: jsonHeaders })
+  page = await response.next()
+  t.deepEqual(page.value.payload, simpleResponseBodies[0], 'Incorrect response body')
+  reset()
+
+  // Query type, second page
+  server.use({ responseBody: simpleResponseBodies[1], responseHeaders: jsonHeaders })
+  page = await response.next()
+  request = server.getCurrentRequest()
+  expectedUrl = '/?Cursor=t1'
+  t.deepEqual(page.value.payload, simpleResponseBodies[1], 'Incorrect response body')
+  t.equal(request.url, expectedUrl, 'Response token does not match next request cursor')
```

Comment:
All of these tests notate "incorrect..." or "does not match..." when a passing test would be correct, or match, etc. As a reminder, the third argument in most of these `tape` methods is the description of the test in question.